src/Kernels/Receiver.cpp

```diff
@@ -216,3 +195,26 @@ double seissol::kernels::ReceiverCluster::calcReceivers(  double time,
   return receiverTime;
 }
 
+void seissol::kernels::ReceiverCluster::allocateData() {
+#ifdef ACL_DEVICE
+  // collect all data pointers to transfer. If we have multiple receivers on the same cell, we make sure to only transfer the related data once (hence, we use the `indexMap` here)
+  deviceIndices.resize(m_receivers.size());
+  std::vector<real*> dofs;
+  std::unordered_map<real*, size_t> indexMap;
+  for (size_t i = 0; i < m_receivers.size(); ++i) {
+    real* currentDofs = m_receivers[i].data.dofs();
+    if (indexMap.find(currentDofs) == indexMap.end()) {
+      // point to the current array end
+      indexMap[currentDofs] = dofs.size();
+      dofs.push_back(currentDofs);
```

Comment:
I see. Wouldn't it more readable with 
```
dofs.push_back(currentDofs);
indexMap[currentDofs] = dofs.back();
```
?

Reply:
Unfortunately, `dofs.back()` is then equal to `currentDofs`—we really want `indexMap[currentDofs]` to point to the index inside the `dofs` vector